src/zope/interface/declarations.py

```diff
@@ -54,6 +54,15 @@ class implements (that instances of the class provides).
 _ADVICE_WARNING = ('The %s API is deprecated, and will not work in Python3  '
                    'Use the @%s class decorator instead.')
 
+def _next_super_class(ob):
+    # When ``ob`` is an instance of ``super``, return
+    # the next class in the MRO that we should actually be
+    # looking at.
+    class_that_invoked_super = ob.__thisclass__
+    mro = class_that_invoked_super.mro()
+    next_class = mro[mro.index(class_that_invoked_super) + 1]
```

Comment:
Again: the `mro` should be determined for `ob.__self_class_`, not `ob.__thisclass__`.